internal/s3client/client.go

```diff
@@ -129,13 +135,20 @@ func (c *S3Client) GetObject(ctx context.Context, bucketName, objectName string)
 			Key:    aws.String(objectName),
 		})
 	if err != nil {
-		return nil, -1, errors.Annotatef(err, "unable to get object %s on bucket %s using S3 client", objectName, bucketName)
+		if err := handleError(err); err != nil {
+			return nil, -1, "", errors.Trace(err)
+		}
+		return nil, -1, "", errors.Annotatef(err, "unable to get object %s on bucket %s using S3 client", objectName, bucketName)
```

Comment:
These annotations all need changing to the form `getting object %s on bucket %s using S3 client`; an error already implying the inability to do something.